fog/view/enclave/impl/src/e_tx_out_store.rs

```diff
@@ -184,24 +184,24 @@ impl<OSC: ORAMStorageCreator<StorageDataSize, StorageMetaSize>> ETxOutStore<OSC>
             );
         }
 
-        // TOOO: Per https://github.com/mobilecoinfoundation/mobilecoin/issues/2965, use a
-        // a constant time comparison function to always copy the same number of bytes.
-        // NOTE: As of right now, this code is not constant time and therefore
-        // blocks the v5 release.
-        // Code to implement:
-        // ```
-        // const LENGTH_TO_COPY: usize = core::cmp::min(FIXED_CIPHERTEXT_LENGTH,
-        //   ValueSize::USIZE - 1);
-        // (&result.ciphertext[..LENGTH_TO_COPY]).copy_from_slice(&value[1..LENGTH_TO_COPY]);
-        // ```
-        let data_end = ValueSize::USIZE - value[0] as usize;
-        let payload = &value[1..data_end];
-        // Use this instead of payload.len() because the slice `len` method isn't
-        // guaranteed to be constant time.
-        let payload_length = data_end - 1;
-        result.ciphertext[0..payload_length].copy_from_slice(payload);
-        result.payload_length = payload_length as u32;
+        // Always copy the same length to ensure constant time. Note that we are not
+        // copying the payload length because this is variable and would
+        // compromise obliviousness.
+        const LENGTH_TO_COPY: usize = core::cmp::min(FIXED_CIPHERTEXT_LENGTH, ValueSize::USIZE - 1);
```

Comment:
I want to make a suggestion:

Instead of taking a `min` of several things with `FIXED_CIPHERTEXT_LENGTH`, let's assert that `Value::USIZE - 1` is less than the `FIXED_CIPHERTEXT_LENGTH`. (Or maybe a `const_assert`.) Since this is all fucked if that assertion isn't right, and then it will make it simpler to see how many bytes are actually copied, and there's no code-path where we silently truncate part of the payload.

Reply:
That sounds good to me. @jcape what do you think?

Reply:
Yep, `const_assert` sounds totally reasonable.